catalog/transformer/catalog-transformer-preview/src/main/java/org/codice/ddf/transformer/preview/PreviewMetacardTransformer.java

```diff
@@ -34,15 +57,62 @@ public BinaryContent transform(Metacard metacard, Map<String, Serializable> argu
     }
 
     String preview = "No preview text available.";
+
     if (metacard.getAttribute(Extracted.EXTRACTED_TEXT) != null
         && metacard.getAttribute(Extracted.EXTRACTED_TEXT).getValue() != null) {
       preview =
           StringEscapeUtils.escapeHtml4(
                   metacard.getAttribute(Extracted.EXTRACTED_TEXT).getValue().toString())
               .replaceAll("[\n|\r]", "<br>");
+    } else if (previewFromMetadata && StringUtils.isNotEmpty(metacard.getMetadata())) {
+      preview =
+          StringEscapeUtils.escapeHtml4(selectPreviewFromMetadata(metacard.getMetadata()))
+              .replaceAll("[\n|\r]", "<br>");
+    }
+
+    if (StringUtils.isNotEmpty(preview)) {
       preview = String.format("<head><meta charset=\"utf-8\"/>%s</head>", preview);
     }
 
     return new BinaryContentImpl(IOUtils.toInputStream(preview));
   }
+
+  private String selectPreviewFromMetadata(String metadata) {
+    XPathFactory xPathFactory;
+    String xPathString = "//*[name()='%s']";
+    String text = "";
+
+    for (String attributeName : textElements) {
+      xPathFactory = XPathFactory.newInstance();
+      try {
+        String xpath = String.format(xPathString, attributeName);
+
+        StringReader metadataReader = new StringReader(metadata);
+        InputSource inputXml = new InputSource(metadataReader);
+
+        Node result =
+            (Node) xPathFactory.newXPath().compile(xpath).evaluate(inputXml, XPathConstants.NODE);
+
+        if (result != null) {
+          text = result.getTextContent().trim();
```

Comment:
❓ Could `getTextContent()` be `null`? This'd throw an `NullPointerException`, if so.

Reply:
This will be resolved by converting the for loop to a stream